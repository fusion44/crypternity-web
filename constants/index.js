export const JWT_TOKEN_VALID = "valid"
export const JWT_TOKEN_VALIDATING = "validating"
export const JWT_TOKEN_INVALID = "invalid"

export const LOCAL_STORAGE_JWT_TOKEN = "jwt_token"
export const LOCAL_STORAGE_USER_ID = "userId"
export const LOCAL_STORAGE_USER_NAME = "userName"
