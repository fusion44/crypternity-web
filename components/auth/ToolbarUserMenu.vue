<template lang="pug">
    div
      div(v-if="getUserLoginState === 'valid'") Hello {{getUserName}}
        v-menu(offset-y)
          v-btn(icon flat slot="activator") 
            v-icon(dark) account_circle
          v-list
            v-list-tile(v-for="item in loggedInMenuItems" :key="item.title" @click="item.click")
              v-list-tile-title {{ item.title }} 
      div(v-else="!loggedIn")
        v-btn(router to="/auth/login"    color="primary") Login
        v-btn(router to="/auth/register" color="primary") Register
</template> 


<script>
import * as consts from "../../constants"
import { mapGetters, mapActions } from "vuex"

export default {
  data() {
    return {
      loggedIn: false,
      loggedInMenuItems: [
        { title: "Settings", click: this.onSettingsClick },
        { title: "Logout", click: this.onLogoutClick }
      ]
    }
  },
  methods: {
    onSettingsClick() {
      console.log("settings")
    },
    onLogoutClick() {
      this.logout()
        .then(res => this.$router.push("/auth/login"))
        .catch(error => console.log(error))
    },
    ...mapActions(["logout"])
  },
  computed: mapGetters(["getUserLoginState", "getUserName"]),
  watch: {
    getUserLoginState(val) {
      return val === consts.JWT_TOKEN_VALID ? this.$router.push("/") : undefined
    }
  }
}
</script>

<style>

</style>
