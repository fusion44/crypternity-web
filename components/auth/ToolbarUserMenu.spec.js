import { shallow, createLocalVue } from "@vue/test-utils"
import Vuex from "vuex"
import ToolbarUserMenu from "./ToolbarUserMenu"
import * as consts from "../../constants"

const localVue = createLocalVue()
localVue.use(Vuex)

describe("ToolbarUserMenu", () => {
  it("renders the correct markup signed out", () => {
    let getters = {
      getUserLoginState: state => consts.JWT_TOKEN_INVALID,
      getUserName: state => ""
    }

    let store = new Vuex.Store({
      getters
    })

    let wrapper = shallow(ToolbarUserMenu, { store, localVue })

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })

  it("renders correct markup signed in", () => {
    let getters = {
      getUserLoginState: state => consts.JWT_TOKEN_VALID,
      getUserName: state => "abcde"
    }

    let store = new Vuex.Store({
      getters
    })

    let wrapper = shallow(ToolbarUserMenu, { store, localVue })

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })
})
