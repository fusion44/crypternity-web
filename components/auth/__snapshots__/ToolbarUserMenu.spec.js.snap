// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarUserMenu renders correct markup signed in 1`] = `
<div>
  <div>Hello abcde
    <v-menu offset-y="offset-y">
      <v-btn slot="activator" icon="icon" flat="flat">
        <v-icon dark="dark">account_circle</v-icon>
      </v-btn>
      <v-list>
        <v-list-tile>
          <v-list-tile-title>Settings </v-list-tile-title>
        </v-list-tile>
        <v-list-tile>
          <v-list-tile-title>Logout </v-list-tile-title>
        </v-list-tile>
      </v-list>
    </v-menu>
  </div>
</div>
`;

exports[`ToolbarUserMenu renders the correct markup signed out 1`] = `
<div>
  <div>
    <v-btn router="router" to="/auth/login" color="primary">Login</v-btn>
    <v-btn router="router" to="/auth/register" color="primary">Register</v-btn>
  </div>
</div>
`;
